### PR TITLE
create base for readthedocs sphinx site

### PR DESCRIPTION
**Pull request by clintonsteiner**
Generate with
```bash
sphinx-build -b html ./source ./build
```
or
```
make html
```

### CODE REVIEW

**Comment by clintonsteiner** [comment]
Linking [https://github.com/akaihola/darker/issues/78](url)

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
```
Comment (akaihola) [inline Makefile:1]: Would it be more modern to use [just] and `justfile`?

[just]: https://just.systems/man/en/

Comment (akaihola) [reply Makefile:1]: Ah, but Sphinx has `Makefile` support out of the box, and probably no support for `just`...

File: source/index.rst [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+   :target: https://github.com/akaihola/darker/blob/master/LICENSE.rst
+.. |pypi-badge| image:: https://img.shields.io/pypi/v/darker
+   :alt: Latest release on PyPI
+   :target: https://pypi.org/project/darker/
+.. |downloads-badge| image:: https://pepy.tech/badge/darker
+   :alt: Number of downloads
+   :target: https://pepy.tech/project/darker
+.. |black-badge| image:: https://img.shields.io/badge/code%20style-black-000000.svg
+   :alt: Source code formatted using Black
+   :target: https://github.com/psf/black
```
Comment (akaihola) [inline source/index.rst:20]: Why not de-duplicate these, too, using `:start-at:` and `:end-before:`?

File: source/index.rst [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+.. |changelog-badge| image:: https://img.shields.io/badge/-change%20log-purple
+   :alt: Change log
+   :target: https://github.com/akaihola/darker/blob/master/CHANGES.rst
+
+What is Darker?
+===============
+
+.. include:: ../README.rst
+   :start-after: What?
+   :end-before: Why?
```
Comment (akaihola) [inline source/index.rst:30]: Could we copy the titles as well, like:
```suggestion
.. include:: ../README.rst
   :start-at: What?
   :end-before: Why?
```

Comment (akaihola) [reply source/index.rst:30]: Or is it that we don't get proper references and tables of content if the headings are not actually literally present in the Sphinx document?

**Review by akaihola — CHANGES_REQUESTED** [review]
In `setup.cfg`, we should probably add `sphinx` to either `darker[release]` extra dependencies, or to a new `darker[docs]` extra.

File: Makefile (hunk 1 above)
Comment (clintonsteiner) [reply Makefile:1]: I'll be honest this is the first I've heard of just and justfile, I do like the support out of the box and know how this works with make if you're good with that

File: source/index.rst (hunk 2 above)
Comment (clintonsteiner) [reply source/index.rst:20]: image
Getting errors doing that way

File: source/index.rst (hunk 3 above)
Comment (clintonsteiner) [reply source/index.rst:30]: Right - we don't have anything to reference to doing that

**Comment by clintonsteiner** [comment]
https://github.com/akaihola/darker/pull/766 
Think this is probably a better approach, thoughts?

**Comment by akaihola** [comment]
> #766 Think this is probably a better approach, thoughts?

#766 is indeed way simpler and adds minimal changes. It's a better approach as long as we choose to stick with a single page of documentation.

I'm trying to think of any benefits from full-fledged multi-page documentation with a more focused README. The change log and contribution guide could then of course be rendered as part of the documentation pages, and maybe more technical detail included as a separate page.

What do you think about the trade-offs between the two approaches?

**Comment by clintonsteiner** [comment]
> > #766 Think this is probably a better approach, thoughts?
> 
> 
> 
> #766 is indeed way simpler and adds minimal changes. It's a better approach as long as we choose to stick with a single page of documentation.
> 
> 
> 
> I'm trying to think of any benefits from full-fledged multi-page documentation with a more focused README. The change log and contribution guide could then of course be rendered as part of the documentation pages, and maybe more technical detail included as a separate page.
> 
> 
> 
> What do you think about the trade-offs between the two approaches?

I think choose the simple option until theres a reason to do the hard one

**Pull request closed without merging**